File: utils/src/main/java/io/github/aemogie/timble/utils/logging/LogManager.java
package io.github.aemogie.timble.utils.logging;

public class LogManager {
	private LogManager() {}
	
	private static Logger logger = null;
	
	private static Logger createLogger() {
		logger = new Logger();
		return logger;
	}
	
	public static boolean nullifyLogger() {
		if (logger.destroy()) {
			logger = null;
			return true;
		}
		return false;
	}
	
	public static Logger getLogger() {
		return logger == null ? createLogger() : logger;
	}
}

File: utils/src/main/kotlin/io/github/aemogie/timble/utils/logging/LogManager.kt
package io.github.aemogie.timble.utils.logging

object LogManager {
	private var logger: Logger? = null
	private fun createLogger(): Logger? {
		logger = Logger()
		return logger
	}
	
	@JvmStatic fun nullifyLogger(): Boolean {
		if (logger!!.destroy()) {
			logger = null
			return true
		}
		return false
	}
	
	@JvmStatic fun getLogger(): Logger? {
		return if (logger == null) createLogger() else logger
	}
}